### DISCUSSION TITLE
Threads id

### DISCUSSION THREAD

**Discussion by Tonaion02** [rayon-rs/rayon#1202] [Q&A]
`    rayon::ThreadPoolBuilder::new().
    num_threads(*NUM_THREADS).
    start_handler(|real_thread_id| {

        println!("Thread id {real_thread_id}");

    }).
    build_global().
    unwrap();`

I noticed that the thread ids passed to start_handler are sequential (0, 1, 2, 3....) and starts from zero.
Is this something casual or is something standard?

**Comment by cuviper** [comment]
The parameter is documented as a thread **index**, and that's specific to the pool, so yes it will be `0..num_threads`. Note that start handlers will be invoked in parallel though, so you might not "see" them sequentially in a global sense.

It will not have any correlation to `std::thread::current().id()` or the operating system `gettid`, etc.
